### PR TITLE
Late escape Query blocks

### PR DESCRIPTION
**Pull request by getdave**
## Description
This is not a security problem. This PR simply moves [escaping of all PHP output to be as "late" as possible](https://developer.wordpress.org/themes/theme-security/data-sanitization-escaping/#:~:text=esc_textarea(%20%24text%20)%3B%20%3F%3E%3C/textarea%3E-,Tip%3A,Output%20escaping%20should%20occur%20as%20late%20as%20possible.,-Top%20%E2%86%91). This means we avoid escaping variables until they are output in the HTML markup.

This is a WP Core best practice.

## How has this been tested?
- add both blocks
- check functionality is "as was"
- check all tests pass

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/block-library/src/query-title/index.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ function render_block_core_query_title( $attributes ) {
 		'<%1$s %2$s>%3$s</%1$s>',
 		$tag_name,
 		$wrapper_attributes,
-		$title
+		esc_html( $title )
```
Comment (ntsekouras) [inline packages/block-library/src/query-title/index.php]: We don't need this, as `$title` is coming from [get_the_archive_title](https://developer.wordpress.org/reference/functions/get_the_archive_title/) which applies filters and html is allowed. The default implementation actually contains a `span`.

**Review by ntsekouras — COMMENTED** [review]
Thanks for the PR Dave!

Without the change for `Query Title` this is good to go!

File: packages/block-library/src/query-title/index.php (hunk 1 above)
Comment (getdave) [reply packages/block-library/src/query-title/index.php]: Ok thanks - done!

**Review by ntsekouras — APPROVED** [review]

**Comment by getdave** [comment]
🥳

**Pull request merged**